### PR TITLE
fix httpReadData will block one http thread until request finished

### PR DESCRIPTION
**Pull request by jianqingdu**
原来的代码效果和阻塞式socket一样，会在http处理线程里一直阻塞到读取一次HTTP请求的所有数据为止

### CODE REVIEW

File: src/plugins/http/src/httpServer.c [hunk 1]
```diff
@@ -400,7 +400,7 @@ static bool httpReadData(HttpContext *pContext) {
     } else if (nread < 0) {
       if (errno == EINTR || errno == EAGAIN || errno == EWOULDBLOCK) {
         httpDebug("context:%p, fd:%d, read from socket error:%d, wait another event", pContext, pContext->fd, errno);
-        continue;  // later again
+        return false;
```
Comment (winshining) [inline src/plugins/http/src/httpServer.c:406]: Excellent!

**Review by winshining — APPROVED** [review]
LGTM

File: src/plugins/http/src/httpServer.c (hunk 1 above)
Comment (winshining) [reply src/plugins/http/src/httpServer.c:406]: @jianqingdu I have discussed with @dapan1121 and we drew a conclusion: if the errno is **EINTR**, the read should immediately continue, while if the errno is EAGAIN or EWOULDBLOCK, return false.

Comment (jianqingdu) [reply src/plugins/http/src/httpServer.c:406]: @winshining I agree with you, and make the change,  please recheck the PR

**Review by winshining — APPROVED** [review]
Good job!

**Review by dapan1121 — APPROVED** [review]

**Pull request merged**